source/opencascade/boundary_lib.cc

```diff
@@ -233,13 +233,13 @@ namespace OpenCASCADE
   ArclengthProjectionLineManifold<dim,spacedim>::pull_back(const Point<spacedim> &space_point) const
   {
     double t (0.0);
-#ifdef DEBUG
     ShapeAnalysis_Curve curve_analysis;
     gp_Pnt proj;
-    Assert(curve_analysis.Project(curve->GetCurve(), point(space_point), tolerance, proj, t, true)
-           < tolerance*length, ExcPointNotOnManifold(space_point));
+#ifdef DEBUG
+    Assert(curve_analysis.Project(curve->GetCurve(), point(space_point), tolerance, proj, t, true) <
+           tolerance*length, ExcPointNotOnManifold(space_point));
 #else
-    (void)space_point;
+    curve_analysis.Project(curve->GetCurve(), point(space_point), tolerance, proj, t, true);
```

Comment:
Ah, can we avoid this somehow? I.e., just execute the code always and only test its return value in the Assert. Executing instructions with side effects inside Assert statements is bound to be confusing, as you yourself have just found out.